### PR TITLE
Earliest year fix and unit tests

### PR DESCRIPTION
**Pull request by weilu**
- Hide health nav link until visuals and narratives are populated
- Fix filter_country_sort_year to update earliest_year with added unit tests
- Add unit test run instructions to README
- Add GitHub action to run tests on PR & merge to main

### CODE REVIEW

**Comment by yukinko-iwasaki** [comment]
@weilu 
Do you know what senario this PR is fixing? 
e.g for which country for which figure?

**Comment by weilu** [comment]
> @weilu Do you know what senario this PR is fixing? e.g for which country for which figure?

The default is Albania, but if you were to switch to any country that doesn't have 2010 as the earliest year like Tunisia, it will throw an error and not display the graph.

File: utils.py [hunk 1]
```diff
@@ -19,8 +19,13 @@ def filter_country_sort_year(df, country):
     """
     df = df.loc[df["country_name"] == country]
 
-    df = df[df.year >= START_YEAR]
+    df = df[df.year >= start_year]
+    if not df.empty:
+        earliest_year = df['year'].min()
```
Comment (yukinko-iwasaki) [inline utils.py:24]: @weilu 
Although I do think it is nice to have the empty check here,
should we also fix the aggregate output data here?

https://github.com/dime-worldbank/mega-boost/blob/a2b9c7baa4c55be8877dc72cc29791674e48c03e/cross_country_aggregate_dlt.py#L114-L122

Here, we are joining the expenditure data and the poverty data. The earliest year is based on the expenditure data, but when joined, the 'earliest year' may not be present in the poverty data, causing some inconsistencies.

 Should we update the earliest year for the resulting aggregate data in boost repo?

Comment (weilu) [reply utils.py:24]: It's a right join where the expenditure data is on the right so all rows from the expenditure data will be retained even if some poverty data is missing for earlier years, so it should be fine. Am I missing something?

Comment (yukinko-iwasaki) [reply utils.py:24]: Got it! I think I was confusing myself with the fix. I am going to approve shortly!

**Review by yukinko-iwasaki — APPROVED** [review]

**Pull request merged**